File: Backend/Travle.Model/Responses/DestinationReviewResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// A rating + comment left on a destination, in the shape used by the reviews list, the author's own
    /// review, and the admin moderation table. The destination and both user references are flattened to
    /// names (never raw ids on screen). Removed reviews are excluded from public reads and rating
    /// aggregates; the moderation view opts them in and reads the removal audit.
    /// </summary>
    public class DestinationReviewResponse
    {
        public int Id { get; set; }

        public int DestinationId { get; set; }
        public string DestinationName { get; set; } = string.Empty;

        public int UserId { get; set; }
        public string Username { get; set; } = string.Empty;
        public string AuthorName { get; set; } = string.Empty;

        public int Rating { get; set; }
        public string? Comment { get; set; }

        public bool IsRemoved { get; set; }
        public int? RemovedByUserId { get; set; }
        public string? RemovedByUsername { get; set; }
        public string? RemovalReason { get; set; }

        public DateTime CreatedAt { get; set; }
        public DateTime? ModifiedAt { get; set; }
    }
}

File: UI/travle_core/lib/src/models/destination_review_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'destination_review_response.g.dart';

/// A rating + comment left on a destination (mirrors the backend
/// `DestinationReviewResponse`). The destination and both user references are
/// flattened to names — never raw ids shown. Removed reviews only reach admins.
@JsonSerializable()
class DestinationReviewResponse {
  DestinationReviewResponse({
    required this.id,
    required this.destinationId,
    required this.destinationName,
    required this.userId,
    required this.username,
    required this.authorName,
    required this.rating,
    required this.isRemoved,
    required this.createdAt,
    this.comment,
    this.removedByUserId,
    this.removedByUsername,
    this.removalReason,
    this.modifiedAt,
  });

  final int id;

  final int destinationId;
  final String destinationName;

  final int userId;
  final String username;
  final String authorName;

  final int rating;
  final String? comment;

  final bool isRemoved;
  final int? removedByUserId;
  final String? removedByUsername;
  final String? removalReason;

  final DateTime createdAt;
  final DateTime? modifiedAt;

  factory DestinationReviewResponse.fromJson(Map<String, dynamic> json) =>
      _$DestinationReviewResponseFromJson(json);

  Map<String, dynamic> toJson() => _$DestinationReviewResponseToJson(this);
}
